arisculala/backend-coding-test
# load-testing tasks code changes

* Added forever in the project to run/stop the application in a daemon
	- start
		`npm run start:forever`
	- stop
		`npm run stop:forever`
	- to check the forever running process (you will be able to see the details of the spin up daemon)
		`forever list`

* Implemented artillery for load testing of the API's
	- run load testing command
		`npm run test:load`


1. Clone the remote repository backend-coding-test in your local machine (using the release tag load-testing)
        $ git clone https://github.com/arisculala/backend-coding-test.git --branch load-testing

2. Go inside the cloned backend-coding-test project
        $ cd backend-coding-test

3. Ensure node (>8.6 and <= 10) and npm are installed

4. Run
        $ npm install

5. Spin up a daemon to run the application
		$ npm run start:forever

6. Run application load testing using artillery
		$ npm run test:load

7. After the load testing, stop the daemon that was spin up by forever
		$ npm run stop:forever
